features/cgal.prf

```diff
@@ -9,7 +11,14 @@ CGAL_DIR = $$(CGALDIR)
   message("CGAL location: $$CGAL_DIR")
 }
 
-LIBS += -lCGAL -lmpfr -lgmp
+exists($$CGAL_DIR/include/CGAL/compiler_config.h)|exists(/usr/include/CGAL/compiler_config.h) {
+  LIBS += -lCGAL -lmpfr -lgmp
+} else {
+  # CGAL header-only
+  DEFINES += CGAL_HEADER_ONLY
+  LIBS += -lmpfr -lgmp
+  message("Using CGAL header-only")
+}
```

Comment:
@t-paul Do you see the error in that logic? It seems the message "Using CGAL header-only" is still displayed on Ubuntu 16.04.

Reply:
Looks like on Ubuntu 16.04 the header is installed in `/usr/include/x86_64-linux-gnu/CGAL/compiler_config.h`. I'm not sure how to check that in a portable way.

Reply:
We could try to parse the content of `<CGAL/version.h>` using qmake, with `$$cat` and `$$find`, but that will probably be tedious.

Could the non-header-only mode of CGAL be triggered by a qmake config variable, set by the user?